contracts/PvpArena.sol

```diff
@@ -467,6 +500,110 @@ contract PvpArena is Initializable, AccessControlUpgradeable {
         return shieldIDs;
     }
 
+    ///@dev function to update the rank of the winner
+    function processWinner(uint256 winnerID) internal {
+        uint256 winnerPoints = characterRankingPoints[winnerID];
+        uint8 tier = getArenaTier(winnerID);
+        uint256[] storage winnerTier = _rankingByTier[tier];
+        uint256 winnerPosition;
+        bool winnerFound;
+
+        // check if winner is withing the top 4
+        for (uint8 i = 0; i < winnerTier.length; i++) {
+            if (winnerID == winnerTier[i]) {
+                winnerPosition = i;
+                winnerFound = true;
+                break;
+            }
+        }
+        // if he is found, compare him to the lower index positions
+        if (winnerFound) {
+            winnerPosition = winnerPosition;
+        }
+        // else, compare it to the 4th one, if he is higher then replace the position and start the loop.
+        else if (
+            winnerPoints >=
+            characterRankingPoints[winnerTier[winnerTier.length - 1]]
+        ) {
+            winnerPosition = winnerTier[winnerTier.length - 1];
+            winnerTier[winnerTier.length - 1] = winnerID;
+        }
+        for (winnerPosition; winnerPosition >= 0; winnerPosition--) {
+            if (winnerPosition <= 0) {
+                break;
+            }
+            if (
+                characterRankingPoints[winnerTier[winnerPosition]] >=
+                characterRankingPoints[winnerTier[winnerPosition - 1]]
+            ) {
+                uint256 newPosition = winnerTier[winnerPosition - 1];
+                winnerTier[winnerPosition - 1] = winnerTier[winnerPosition];
+                winnerTier[winnerPosition] = newPosition;
+            } else {
+                break;
+            }
+        }
+    }
+
+    ///@dev function to update the rank of the loser
+    function processLoser(uint256 loserID) internal {
+        uint256 loserPoints = characterRankingPoints[loserID];
+        uint8 tier = getArenaTier(loserID);
+        uint256[] storage loserTier = _rankingByTier[tier];
+        uint256 loserPosition;
+        bool loserFound;
+
+        // check if the loser is within the top 4 players
+        for (uint8 i = 0; i < loserTier.length; i++) {
+            if (loserID == loserTier[i]) {
+                loserPosition = i;
+                loserFound = true;
+                break;
+            }
+        }
+        // if he is found, compare him to the upper positions and replace the rank accordingly
```

Comment:
Wouldn't it be the "lower" positions? In a ranking, higher means better so this might be confusing